python/pyarrow/__init__.py

```diff
@@ -151,6 +151,29 @@ def parse_git(root, **kwargs):
 
 import pyarrow.types as types
 
+# GPU support
+#
+try:
+    import pyarrow.lib_gpu as _lib_gpu
+except ImportError as _msg:
+    _msg = str(_msg)
+    if _msg.startswith('No module named') and 'lib_gpu' in _msg:
+        has_gpu_support = False
+    else: # do not silence ImportError in case of any bugs
+        raise
+else:
+    has_gpu_support = True
+
+if has_gpu_support:
+    from pyarrow.lib_gpu \
+        import (CudaDeviceManager, CudaContext, CudaIpcMemHandle,
+                CudaBuffer, CudaHostBuffer, CudaBufferReader, CudaBufferWriter,
+                allocate_cuda_host_buffer, cuda_serialize_record_batch,
+                cuda_read_message, cuda_read_record_batch,
+                )
```

Comment:
We aren't injecting optional components into the `pyarrow.*` namespace. How about we create a `pyarrow.cuda` namespace and put these symbols there?

Reply:
OK. This also means that one get rid of `cuda` prefixes, that is, a user would use
```
pa.cuda.Buffer, pa.cuda.HostBuffer, ...
pa.cuda.read_message cuda.read_record_batch, ...
```
instead of `pa.cuda.CudaBuffer`, etc.

Reply:
Created pyarrow.cuda and removed cuda prefix everywhere except for CudaBuffer because in docs it would be confusing to use Buffer without relating it to `pa.Buffer` or `pa.cuda.Buffer`.